### ISSUE TITLE
Specifying a metric TTL as a large number as a string causes crash

### ISSUE THREAD

**Issue by kbuchsicf** [sensu/sensu#1871]
We have been, in ignorance, submitting POST requests to Sensu server with metric data and TTL given as a string. When the TTL was given a larger number, as a string, it caused a crash in sensu-server and redis. 

## Expected Behavior
Sensu server & redis should not crash. 

## Current Behavior
Crashing, requiring restart of sensu server and redis

## Possible Solution
Correct interpretation of TTL as a string or reject the metric all together. If the magnitude of the TTL is really the issue, then generate an appropriate response. No matter what, don't accept a metric which will then cause a crash.

## Steps to Reproduce (for bugs)
1. Submit a metric with "ttl": "3672000"    ; that is 42 days

Contrast this with "ttl": "648000"   ; that is 7 days
which was not causing a crash

## Context
Crash halted operation of sensu server and loss of data.

## Your Environment
We are running with Docker containers; with these images:
```# docker images
REPOSITORY                      TAG                 IMAGE ID            CREATED             SIZE
sensudockerstack_sensu-server   latest              9e1c63af99c5        7 months ago        640MB
<none>                          <none>              45fd29784c15        7 months ago        638MB
sensudockerstack_sensu-client   latest              75eff35c5b9a        7 months ago        1.02GB
<none>                          <none>              343393fd0739        7 months ago        638MB
sensudockerstack_sensu-api      latest              b9ad91385d6c        7 months ago        616MB
<none>                          <none>              69835694ca7e        7 months ago        1.02GB
<none>                          <none>              0271e5941137        7 months ago        638MB
<none>                          <none>              b7dafb8bd204        7 months ago        616MB
rabbitmq                        3-management        046c07b70859        7 months ago        124MB
uchiwa/uchiwa                   latest              68d89653a1dc        8 months ago        346MB
hopsoft/graphite-statsd         latest              6a63c117b7c9        8 months ago        870MB
grafana/grafana                 latest              1d2e2705fd26        8 months ago        303MB
redis                           alpine              5c7823413ff7        8 months ago        27.5MB
uberamd/sensu-ubuntu            1.0.2               2b2ce5e88a45        9 months ago        207MB

# docker ps
CONTAINER ID        IMAGE                           COMMAND                  CREATED             STATUS              PORTS                                                                                                                                    NAMES
382fca81ae92        sensudockerstack_sensu-server   "/tmp/run.sh"            7 months ago        Up 6 hours                                                                                                                                                   sensu-server
765152976f99        sensudockerstack_sensu-client   "/tmp/run.sh"            7 months ago        Up 7 months         3030/tcp                                                                                                                                 sesnu-client
0285c633bcf3        uchiwa/uchiwa                   "/go/bin/uchiwa -c..."   7 months ago        Up 7 months         0.0.0.0:3000->3000/tcp                                                                                                                   sensu-uchiwa
a206f93d6354        sensudockerstack_sensu-api      "/tmp/run.sh"            7 months ago        Up 5 months         0.0.0.0:4567->4567/tcp                                                                                                                   sensu-api
b457d9aeffb1        grafana/grafana                 "/run.sh"                7 months ago        Up 4 months         0.0.0.0:3333->3000/tcp                                                                                                                   sensudockerstack_grafana_1
65496fb38d30        rabbitmq:3-management           "docker-entrypoint..."   7 months ago        Up 7 months         4369/tcp, 5671/tcp, 0.0.0.0:5672->5672/tcp, 15671/tcp, 25672/tcp, 0.0.0.0:15672->15672/tcp                                               sensu-rabbitmq
3eb2296886ec        hopsoft/graphite-statsd         "/sbin/my_init"          7 months ago        Up 7 months         0.0.0.0:80->80/tcp, 0.0.0.0:2003-2004->2003-2004/tcp, 0.0.0.0:2023-2024->2023-2024/tcp, 0.0.0.0:8126->8126/tcp, 0.0.0.0:8125->8125/udp   sensudockerstack_graphite_1
60d63ab91821        redis:alpine                    "docker-entrypoint..."   7 months ago        Up 6 hours          6379/tcp                                                                                                                                 sensu-redis
```
* Sensu version used: 1.1.0
[version-manifest.txt](https://github.com/sensu/sensu/files/2060071/version-manifest.txt)

* Active Sensu extensions: N/A
* Operating System and version: CentOS Linux release 7.4.1708 (Core)
* Transport and runtime versions (e.g. RabbitMQ 3.6.2 on Erlang 18.2): See container image versions above.

### Traceback

```{"timestamp":"2018-05-31T14:31:45.452786+0000","level":"warn","message":"reconnecting to redis"}
/opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.2/lib/sensu/server/process.rb:1083:in `>=': comparison of Integer with String failed (ArgumentError)
        from /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.2/lib/sensu/server/process.rb:1083:in `block (3 levels) in determine_stale_check_results'
        from /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-redis-2.2.0/lib/sensu/redis/client.rb:337:in `dispatch_response'
        from /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-redis-2.2.0/lib/sensu/redis/client.rb:359:in `parse_response_line'
        from /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-redis-2.2.0/lib/sensu/redis/client.rb:387:in `receive_data'
        from /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/eventmachine-1.2.2/lib/eventmachine.rb:194:in `run_machine'
        from /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/eventmachine-1.2.2/lib/eventmachine.rb:194:in `run'
        from /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.2/lib/sensu/server/process.rb:33:in `run'
        from /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.2/exe/sensu-server:10:in `<top (required)>'
        from /opt/sensu/bin/sensu-server:22:in `load'
        from /opt/sensu/bin/sensu-server:22:in `<main>'
Setting plugin permissions
Running sensu config:
{
  "redis": {
    "port": 6379,
    "host": "sensu-redis",
    "password": "password",
    "reconnect_on_error": true
  },
  "rabbitmq": {
    "host": "sensu-rabbitmq",
    "port": 5672,
    "vhost": "sensu",
    "user": "sensu",
    "password": "password"
  },
  "api": {
    "port": 4567,
    "host": "sensu-api",
    "user": "sensu",
    "password": "password"
  }
}
```
After traceback

**Comment by portertech** [comment]
Settings validator should be catching this: https://github.com/sensu/sensu-settings/blob/6609d548b2dfad4de7a474e6eb5692732abd1f47/lib/sensu/settings/validators/check.rb#L131-L133

**Comment by portertech** [comment]
Perhaps the client socket is not validating TTL.

**Comment by kbuchsicf** [comment]
I am submitting a metric outside of the client, and yes, I had that code misconfigured. So, my bad - full disclosure. But... The behavior I observed is that smaller TTLs up to 1 week + 12 hours, submitted as a string were properly accepted and processed with no ill effects. When I tried a TTL of 42 days as a string, it started crashing the Sensu server. Of course, changing an integer instead of a string resolved the issue. However, the crashing Sensu was my concern. I see the code you reference, but I think somehow that must not have been relevant in the case I had, because smaller numbers as strings were working properly.
